generals/game.py

```diff
@@ -155,13 +155,13 @@ def step(self, actions: Dict[str, np.ndarray]):
         """
         done_before_actions = self.is_done()
 
-        # Process whether agents want to pass the turn, and calculate
-        # intended amount of army to move (all available or split)
+        # Process validity of moves, whether agents want to pass the turn,
+        # and calculate intended amount of army to move (all available or split)
         moves = {}
         for agent, move in actions.items():
             pass_turn, i, j, direction, split_army = move
-            # Skip if wants to pass the turn
-            if pass_turn == 1:
+            # Skip if the move is not valid or the agent wants to pass the turn
+            if self.action_mask(agent)[i, j, direction] == 0 or pass_turn == 1:
```

Comment:
do I understand correctly that with this check, invalid moves are handled gently (game should not crash) ?